python/strings_udf/strings_udf/lowering.py

```diff
@@ -0,0 +1,271 @@
+    result = context.compile_internal(
+        builder,
+        call_len_string_view,
+        nb_signature(size_type, _STR_VIEW_PTR),
+        (sv_ptr,),
+    )
+
+    return result
+
+
+def create_binary_string_func(binary_func, retty):
+    def deco(cuda_func):
+        @cuda_lower(binary_func, string_view, string_view)
+        def binary_func_impl(context, builder, sig, args):
+            lhs_ptr = builder.alloca(args[0].type)
+            rhs_ptr = builder.alloca(args[1].type)
+
+            builder.store(args[0], lhs_ptr)
+            builder.store(args[1], rhs_ptr)
+            result = context.compile_internal(
+                builder,
+                cuda_func,
+                nb_signature(retty, _STR_VIEW_PTR, _STR_VIEW_PTR),
+                (lhs_ptr, rhs_ptr),
+            )
+
+            return result
+
+        return binary_func_impl
+
+    return deco
+
+
+@create_binary_string_func(operator.contains, types.boolean)
+def string_view_contains_impl(st, substr):
+    return _string_view_contains(st, substr)
+
+
+@create_binary_string_func(operator.eq, types.boolean)
+def string_view_eq_impl(st, rhs):
+    return _string_view_eq(st, rhs)
+
+
+@create_binary_string_func(operator.ne, types.boolean)
+def string_view_ne_impl(st, rhs):
+    return _string_view_ne(st, rhs)
+
+
+@create_binary_string_func(operator.ge, types.boolean)
+def string_view_ge_impl(st, rhs):
+    return _string_view_ge(st, rhs)
+
+
+@create_binary_string_func(operator.le, types.boolean)
+def string_view_le_impl(st, rhs):
+    return _string_view_le(st, rhs)
+
+
+@create_binary_string_func(operator.gt, types.boolean)
+def string_view_gt_impl(st, rhs):
+    return _string_view_gt(st, rhs)
+
+
+@create_binary_string_func(operator.lt, types.boolean)
+def string_view_lt_impl(st, rhs):
+    return _string_view_lt(st, rhs)
+
+
+@create_binary_string_func("StringView.startswith", types.boolean)
+def string_view_startswith_impl(sv, substr):
+    return _string_view_startswith(sv, substr)
+
+
+@create_binary_string_func("StringView.endswith", types.boolean)
+def string_view_endswith_impl(sv, substr):
+    return _string_view_endswith(sv, substr)
+
+
+@create_binary_string_func("StringView.count", size_type)
+def string_view_count_impl(st, substr):
+    return _string_view_count(st, substr)
+
+
+@create_binary_string_func("StringView.find", size_type)
+def string_view_find_impl(sv, substr):
+    return _string_view_find(sv, substr)
+
+
+@create_binary_string_func("StringView.rfind", size_type)
+def string_view_rfind_impl(sv, substr):
+    return _string_view_rfind(sv, substr)
+
+
+def create_unary_identifier_func(id_func):
+    def deco(cuda_func):
+        @cuda_lower(id_func, string_view)
+        def id_func_impl(context, builder, sig, args):
+            str_ptr = builder.alloca(args[0].type)
+            builder.store(args[0], str_ptr)
+            tbl_ptr = context.get_constant(
+                types.int64, character_flags_table_ptr
```

Comment:
This appears to be the only use of the `character_flags_table_ptr`, which is why I made the suggestion in the pxd file to see if we could remove this entirely. I don't see why it needs to be a (required) parameter for the C++ functions instead of just something we pass through.

Reply:
It needs to be resolved at runtime and not in device code. It is a singleton but it cannot be initialized as a global variable because it requires CUDA calls to resolve it. The infrastructure here must call it and pass the pointer down to the kernels.

Reply:
Should we add a comment here (or somewhere) about this? 
Not that I don't mind re-explaining it each time it comes up. :-)

Reply:
Oh yes I think that would be helpful.